client/routes/workflow/components/cytoscape-graph.vue

```diff
@@ -0,0 +1,237 @@
+<template>
+  <div id="cytoscape">
+    <div ref="cy" id="cy"></div>
+  </div>
+</template>
+
+<script>
+import dagre from "cytoscape-dagre";
+import { getEventConnections } from "../helpers/get-event-connections";
+import graphStyles from "../helpers/graph-styles";
+import store from "../../../store/index";
+import cytoscape from "cytoscape";
+import omit from "lodash-es/omit";
+
+cytoscape.use(dagre);
+
+export default {
+  name: "cytoscape-graph",
+  props: ["events"],
+  data() {
+    return {
+      nodes: [],
+      edges: [],
+      styles: graphStyles,
+      parentArray: []
+    };
+  },
+  watch: {
+    selectedEvent(id) {
+      if (id) this.selectNode(id);
+    }
+  },
+  methods: {
+    zoomToNode(node) {
+      let zoom = 1.1,
+        bb = node.boundingBox(),
+        w = cy.width(),
+        h = cy.height(),
+        pan = {
+          x: (w - zoom * (bb.x1 + bb.x2)) / 2,
+          y: (h - zoom * (bb.y1 + bb.y2)) / 2
+        };
+
+      //Pan the graph to view node
+      cy.animate({
+        zoom: 1.1,
+        pan: pan
+      });
+    },
+    updateChildBtn(node) {
+      let nodeData = node.data();
+      if (nodeData.childRoute) {
+        store.commit("childRoute", {
+          route: nodeData.childRoute,
+          btnText: "To child"
+        });
```

Comment:
It's generally not a good idea to directly call store from a component. The reason is that say you wanted to reuse this component elsewhere or maybe you want a styleguide to include a basic component, it isn't that portable now as it relies on a particular store setup. It is much more flexible if you have a component, a connector and a container such that when all three are used together it connects the state to the component.
I know Vuex does support this method, however its better practice to use `mapMutations` (see [here](https://vuex.vuejs.org/guide/mutations.html#committing-mutations-in-components)) or even better using connect (similar to how redux works, see [here](https://github.com/ktsn/vuex-connect)). If using the connect approach, you can have 1 component, but several different containers all referencing the single component. Then all the wireup logic is made in the connector part of the container.